python/pyspark/sql/types.py

```diff
@@ -1209,6 +1219,12 @@ def __new__(self, *args, **kwargs):
         else:
             raise ValueError("No args or kwargs")
 
+    def __init__(self, *args, **kwargs):
+        if hasattr(self, "__fields__") and "count" in self.__fields__:
```

Comment:
To do this check once per class might be not possible because we can only know the field names during initialise a row.